src/iter.rs

```diff
@@ -25,205 +27,133 @@ where
 }
 
 /// Wraps an iterator to display its progress.
+#[derive(Debug)]
 pub struct ProgressBarIter<T> {
-    it: T,
+    pub(crate) inner: T,
     pub progress: ProgressBar,
 }
 
 impl<S, T: Iterator<Item = S>> Iterator for ProgressBarIter<T> {
     type Item = S;
 
     fn next(&mut self) -> Option<Self::Item> {
-        let next = self.it.next();
+        let item = self.inner.next();
 
-        if next.is_some() {
+        if item.is_some() {
             self.progress.inc(1);
         } else {
             self.progress.finish();
         }
 
-        next
+        item
     }
 }
 
-impl<S, T: Iterator<Item = S>> ProgressIterator for T {
-    fn progress_with(self, progress: ProgressBar) -> ProgressBarIter<Self> {
-        ProgressBarIter { it: self, progress }
+impl<T: ExactSizeIterator> ExactSizeIterator for ProgressBarIter<T> {
```

Comment:
Would you mind splitting this commit into a few separate ones (or separate PRs)? For example, one that moves rayon support to a separate file, one that moves `ProgressBarIter`, one that renames field names, and one with other changes? Keeping everything in the same commit makes it much harder for me to review.

Reply:
Sure thing, it should be easier now.